### DISCUSSION TITLE
How do you set ambient light?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3483] [Q&A] [answered]
Probably a dumb question, but I don't see how to set up ambient light.  `IndirectLight` looks like something much more complicated, and `LightManager` does not seem to handle it.

**Comment by romainguy** [comment, marked as answer]
`IndirectLight` is an ambient light, actually the only type of ambient light you can use. An `IndirectLight` is not set via `LightManager`,  instead you set it on the scene with `Scene::setIndirectLight`.

**Reply by jwwalker** [reply]
OK, but how do you make an `IndirectLight` without a cube map?  I'm looking for something equivalent to the old OpenGL `glLightModelfv( GL_LIGHT_MODEL_AMBIENT, someColor )`.

**Reply by romainguy** [reply]
You can make a solid colored cubemaps, either offline or programmatically. Just fill all the pixel data with a single color. If you want to provide SH all you need is set the first value to the ambient color and everything else to 0. I would however recommend using an actual environment map for better results. You'll get very flat lighting otherwise.

**Reply by jwwalker** [reply]
Thanks!  I knew that using only ambient light would produce a flat effect.  Normally I would combine a little ambient light with one or more directional, point, or spot lights.

**Reply by romainguy** [reply]
Unless you'll move those lights or enable multiple shadows, it's better to just bake them in your environment map. It's not quite the same but it gives great results at a low cost. "Studio" lighting environment are therefore pretty common.
